source/blog/2015-06-12-ember-1-13-0-released.md

```diff
@@ -0,0 +1,578 @@
+{{! app/components/show-full-name }}
+{{if hasBlock}}
+  {{yield fullName}}
+{{else}}
+  {{fullName}}
+{{/if}}
+```
+
+Then these two usages would be valid:
+
+```hbs
+{{! app/index/template.hbs }}
+
+Full name: {{show-full-name firstName=firstName lastName=lastName}}
+
+{{#show-full-name firstName=firstName lastName=lastName as |fullName|}}
+  Full name: {{fullName}}
+{{/show-full-name}}
+```
+
+Additionally, `hasBlockParams` will be true if the component is invoked
+with block params (invoke in block form with `as |someParam|`).
+
+Thanks to [@mmun](https://twitter.com/_mmun) and [@rwjblue](https://twitter.com/rwjblue)
+for implementing this feature.
+
+#### Notable Deprecations in 1.13
+
+In preparation for Ember 2.0, 1.13 introduces many deprecations. These include:
+
+* All view APIs in Ember
+  * `Ember.CoreView`, `Ember.View`, `Ember.CollectionView`, `Ember.ContainerView`
+  * `{{view 'some-helper'}}`
+  * The `{{view}}` keyword for accessing properties on a view
+  * `Ember.Select` and `{{view "select"}}`.
+Options to the `{{#each` helper that trigger a legacy and poorly performing
+* Options to the `{{#each` helper that trigger a legacy and poorly performing
+  legacy layer. These options are: `itemView`, `itemViewClass`, `tagName`, `emptyView` and `emptyViewClass`.
+* The `itemController` argument for `{{#each`.
+* The `bind-attr` helper. Using helpers and HTMLBars-style attribute binding
+  is preferred.
+* Reading `this.get('template')` to check for a yielded block on components.
+  Instead, use the `hasBlock` API.
+* Non-block param `{{with`
+* The `view` and `viewClass` params for `{{outlet}}`
+
+## Ember 2.0 beta
+
+Last November, @wycats and @tomdale published [The Road to Ember 2.0](https://github.com/emberjs/rfcs/pull/15),
+an RFC summarizing Ember's goals for a 2.0 release. At EmberConf, when some of
+these changes were already complete and others not begun, they announced
+our intent to ship 2.0 beta on June 12th.
+
+Together, the features summarized in the 2.0 RFC describe a new way to
+author Ember applications. Model-View-Controller is replaced by
+Model-Route-Component-Service. Two-way bindings are replaced by data down,
+actions up (DDAU).
+
+Ember 2.0 will not introduce the entirety of our improved development model.
+However it will take significant steps in that direction, and allow the
+removal of public APIs that have been difficult to maintain while we
+iterate forward.
+
+Because of the focus on landing migration paths for 1.x codebases in 1.13,
+2.0 will have few new features. Among them are:
+
+#### each-in helper
+
+The `each-in` helper allows the iteration of object properties. For example,
+given this value for `items`:
+
+```js
+let items = {
+  "Item 1": 1234,
+  "Item 2": 3456
+};
+```
+
+The following template will iterate the keys:
+
+```hbs
+{{#each-in items as |key value|}}
+  <p>{{key}}: {{value}}</p>
+{{/each-in}}
+```
+
+Note that this helper is unbound. Adding a new property to `items` will not
```

Comment:
but if `items` is replaced, it updates correct? Merely the properties are not "live"